### DISCUSSION TITLE
Cask screens: correct latest version, wrong checksum

### DISCUSSION THREAD

**Discussion by growf** [Homebrew/discussions#4671] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.1-31-g471e733
ORIGIN: https://github.com/Homebrew/brew
HEAD: 471e733641284b27b7cb98e7329f8efb70fe1dbf
Last commit: 4 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 68b3e3d8cb56466c2c750647b2a8e5caa7f5921c
Core tap last commit: 19 minutes ago
Core tap branch: master
Core tap JSON: 25 Jul 18:22 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.ZPM9pKeG7G/org.xquartz:0
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Cask `screens` is showing the correct latest version (4.12.10b1689789098) but an incorrect checksum:
```
==> Downloading https://updates.edovia.com/com.edovia.screens4.mac/Screens_4.12.10b1689789098.zip
############################################################################################################################# 100.0%
Error: SHA256 mismatch
Expected: edf188f25384d7c39857fec9ce491aedf7176bd6b52ea5b41c5e433285cdfe8c
  Actual: 2b971afaa18076d46eee213b432ad86beb429c64eb97a76a9074c9751ab0887b
    File: /Users/bruce/Library/Caches/Homebrew/downloads/569eb158d1c7f985f45605462aa876098d0e65db3df37b69ad6f8489d6ba83b5--Screens_4.12.10b1689789098.zip
To retry an incomplete download, remove the file above.
```
Given the version is correct (according to [Screen's homepage](https://edovia.com/en/screens-mac/)) what's the right course of action?

### Relevant casks

screens

**Comment by growf** [comment, marked as answer]
I've created a [PR](https://github.com/Homebrew/homebrew-cask/pull/151678) if no one's weirded out by the checksum change.

**Comment by gromgit** [comment]
> Given the version is correct (according to [Screen's homepage](https://edovia.com/en/screens-mac/)) what's the right course of action?

Homebrew guarantees that downloaded assets have the stated checksums _at the time the formula/cask was committed_. If the checksum no longer matches, _ALWAYS_ confirm that the asset was changed intentionally by the upstream developers, instead of blindly assuming that it has not been compromised by some Evil Third Party.

Also confirm that you don't have a proxy in between you and the upstream server, that may have altered downloads for its own reasons. At least one Homebrew user reported checksum mismatches recently, then confirmed that his corporate firewall was mangling certain downloads.
